src/java/org/apache/cassandra/schema/TableMetadata.java

```diff
@@ -196,6 +200,9 @@ public enum Kind
     public final DataResource resource;
     public TableMetadataRef ref;
 
+    public final List<ColumnConstraint> partitionKeyConstraints;
+    public final List<ColumnMetadata> columnsWithConstraints;
```

Comment:
Recommend you hoist up the comment below around "We cache the columns with constraints to avoid iterations over columns" to explain the purpose of this here and why the asymmetry in naming.